arch/risc-v/src/common/riscv_swint.c

```diff
@@ -248,9 +264,21 @@ int riscv_swint(int irq, void *context, void *arg)
         {
           DEBUGASSERT(regs[REG_A1] != 0 && regs[REG_A2] != 0);
           riscv_savefpu(regs);
-          riscv_restorefpu((uintptr_t *)regs[REG_A2]);
           *(uintptr_t **)regs[REG_A1] = (uintptr_t *)regs;
           CURRENT_REGS = (uintptr_t *)regs[REG_A2];
+#ifdef CONFIG_ARCH_ADDRENV
+
+          tcb = this_task();
+
+          /* Make sure that the address environment for the previously
+           * running task is closed down gracefully (data caches dump,
+           * MMU flushed) and set up the address environment for the
+           * new thread at the head of the ready-to-run list.
+           */
+
+          group_addrenv(tcb);
```

Comment:
Yes IRQ stack is in use currently, but @anchao reported similar patch can't fix all issues on arm , does this PR works on RISC-V ?

Reply:
I am testing right now but I had to modify the SW quite a bit to fix comments in #5782 and the SW crashes before this point still

Reply:
Now I got to this point, but the SW still crashes somewhere.

This can be my fault, so I will investigate more. IMO this should be the correct place to swap address environment with the new "save context to stack" approach

Reply:
The crash happens in `riscv_savefpu` 

135                   FSTORE        f0,  REG_F0(a0)
0000000080000a30:   fsd     ft0,264(a0)

It gives illegal instruction fault, which is a bit strange

Reply:
@no1wudi do I need this #5905 also for this change to work ?